File: data/test00.rs
//! Module-level documentation for Coding Agent testing.

use std::collections::HashMap;

/// Manages user session and authentication.
pub struct UserManager {
    pub db_url: String,
}

impl UserManager {
    pub fn new(db_url: &str) -> Self {
        UserManager {
            db_url: db_url.to_string(),
        }
    }

    pub fn get_user_by_id(&self, _user_id: i32) -> Option<HashMap<String, String>> {
        // TODO: Implement database lookup
        None
    }
}

fn main() {
    // パターンマッチング (match) のテスト
    let status = 200;
    match status {
        200 | 201 => println!("Success"),
        _ => println!("Failure"),
    }
}

File: data/test00.py
"""Module-level documentation for Coding Agent testing."""

import os
from typing import List, Optional
DB = "http://localhost"

class UserManager:
    """Manages user session and authentication."""
    default_url = DB
    def __init__(self, db_url: str):
        self.db_url = db_url

    def get_user_by_id(self, user_id: int) -> Optional[dict]:
        # TODO: Implement database lookup
        return None

def main():
    status = 200
    match status:
        case 200 | 201:
            print("Success")
        case _:
            print("Failure")

if __name__ == "__main__":
    main()
